app/ts/simulation/services/SimulationModeEthereumClientService.ts

```diff
@@ -450,16 +450,17 @@ export const getSimulatedCode = async (ethereumClientService: EthereumClientServ
 		input: input,
 		accessList: []
 	} as const
-
-	const simulatedTransactions = await simulateTransactionsOnTopOfSimulationState(ethereumClientService, requestAbortController, simulationState, [getCodeTransaction], { [addressString(GET_CODE_CONTRACT)]: { code: getCodeByteCode() } })
-	const lastResult = simulatedTransactions.at(-1)
-	if (lastResult === undefined) throw new Error('last result did not exist in multicall')
-	if (lastResult.status === 'failure') return { statusCode: 'failure' } as const
-	const parsed = atInterface.decodeFunctionResult('at', lastResult.returnData)
-	return {
-		statusCode: lastResult.status,
-		getCodeReturn: EthereumData.parse(parsed.toString())
-	} as const
+	try {
+		const simulatedTransactions = await simulateTransactionsOnTopOfSimulationState(ethereumClientService, requestAbortController, simulationState, [getCodeTransaction], { [addressString(GET_CODE_CONTRACT)]: { code: getCodeByteCode() } })
+		const lastResult = simulatedTransactions.at(-1)
+		if (lastResult === undefined) throw new Error('last result did not exist in multicall')
+		if (lastResult.status === 'failure') return { statusCode: 'failure' } as const
```

Comment:
The lines that are throws are situations that are unexpected while the `statusCode: failure` are "expected" failure modes?

Reply:
yeah, the thrown ones should never happen if the RPC behaves correctly. The other failure happen on normal cases